File: backend/app/models/appointment.py
from datetime import datetime
from enum import StrEnum

from pydantic import BaseModel


class AppointmentStatus(StrEnum):
    SCHEDULED = "scheduled"
    COMPLETED = "completed"
    CANCELLED = "cancelled"
    NO_SHOW = "no_show"


class AppointmentBase(BaseModel):
    client_id: str
    appointment_datetime: datetime
    service_type: str
    provider: str
    location: str
    status: AppointmentStatus = AppointmentStatus.SCHEDULED
    authorization_id: str | None = None

File: frontend/lib/validation/appointment.ts
import { z } from "zod";

export const appointmentFormSchema = z.object({
  client_id: z.string().min(1, "Select a client"),
  appointment_datetime: z.string().min(1, "Required"),
  service_type: z.string().min(1, "Required"),
  provider: z.string().min(1, "Required"),
  location: z.string().min(1, "Required"),
  status: z.enum(["scheduled", "completed", "cancelled", "no_show"]),
});

export type AppointmentFormValues = z.infer<typeof appointmentFormSchema>;
